src/hdtAssetValidator.cpp

```
@@ -406,6 +406,86 @@ namespace hdt
+	static void enqueueUniqueXMLPath(const std::string& path,
+		std::unordered_set<std::string>& seen,
+		std::vector<std::string>& queue)
+	{
+		if (path.empty())
+			return;
+		auto norm = normalisePath(path);
+		if (seen.insert(norm).second)
+			queue.push_back(path);
+	}
+
+	static std::vector<std::string> collectAllPhysicsXMLPaths()
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)